index.html

```diff
@@ -1049,6 +1049,371 @@ <h2>Evidence</h2>
+				"nonce": "598c63d6",
+				"signatureValue": "BavEll0...W3JT24 = "
+			}
+		}
+	},
+	"proof": { ...
+	}
+}
+
+        </pre>
+
+
+  </section>
+
+  <section>
+   <h3> Delegation through Profiles </h3>
+
+   <p>
+
+Through the use of profiles, the subject delegates a credential to a
+holder, by issuing a new credential in which: the issuer is the subject, 
+the subject is the delegated holder, and the claim contains the property to be delegated. In addition, the delegated holder creates a profile that contains these two credentials, and a relationships property in which one
+relationship refers to the original subject and has the value 'delegate' and the other
+relationship refers to the delegated credential and has the value 'self'.
+
+        <pre class="example nohighlight" title="An example of delegation using a profile">
+{
+	"id": "did:example:ebfeb1c276e12ec211f712ebc6f",
+	"type": ["VerifiableProfile"],
+	"holderid": "did:example:76e12ec21ebhyu1f712ebc6f1z2",
+	"relationships": [{
+			"subjectid": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+			"relationship": "delegate"
+		},
+		{
+			"subjectid": "did:example:76e12ec21ebhyu1f712ebc6f1z2",
+			"relationship": "self"
+		}
+	],
+	"credential": [{
+			"id": "http://example.gov/credentials/3732",
+			"type": ["Credential", "Role"],
+			"issuer": "https://dmv.example.gov",
+			"issued": "2010-01-01",
+			"claim": {
+				"id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+				"role": "Project Manager"
+			},
+			"revocation": {
+				"id": "http://example.gov/revocations/738",
+				"type": "SimpleRevocationList2017"
+			},
+			"proof": {....}
+		},
+		{
+			"id": "some other ID",
+			"type": ["Credential", "DelagatedCredential"],
+			"issuer": "did:example:ebfeb1f712ebc6f1c276e12ec21",
+			"issued": "2010-01-03",
+			"claim": {
+				"id": "did:example:76e12ec21ebhyu1f712ebc6f1z2",
+				"role": "Project Manager"
+			},
+			"proof": {
+				"type": "RsaSignature2018",
+				"created": "2017-06-17T10:03:48Z",
+				"creator": "did:example:ebfeb1f712ebc6f1c276e12ec21/keys/234",
+				"nonce": "d61c4599-0cc2-4479-9efc-c63add3a43b2",
+				"signatureValue": "pYw8XNi1..Cky6Ed = "
+			}
+		}
+	],
+	"proof": [{
+		"type": "RsaSignature2018",
+		"created": "2017-06-18T21:19:10Z",
+		"creator": "did:example:76e12ec21ebhyu1f712ebc6f1z2/keys/2",
+		"nonce": "c0ae1c8e-c7e7-469f-b252-86e6a0e7387e",
+		"signatureValue": "BavEll0/I1..W3JT24 = "
+	}]
+}
+        </pre>
+
+
+   </section>
+  </section>
+
+
+  <section>
+      <h2>Issuer Authorises Holder</h2>
+
+      <p>
+When the issuer wishes to authorise a holder to possess a credential that 
+describes a subject who is not the holder, then the issuer inserts the ID 
+of the holder into the subject's credential, by using the holderid property.
+
+      </p>
+
+    <section>
+      <h3> Holder ID Property</h3>
+
+     <p>
+The holderid property is inserted into a credential by the issuer, when the 
+issuer authorises the holder to possess the credential.
```

Comment:
If it can be done in OCAP it can be done with VCs :-)

Reply:
This has been said several times (and [discussed elsewhere](https://github.com/w3c-ccg/ocap-ld/issues/6#issuecomment-379823301)), but it isn't the case.  Verifiable Credentials are extremely useful for *a different purpose than ocaps*.  VCs can be used to "extend a record" of another entity by saying all sorts of things about them.  Alice is tall, Alice has blue eyes, Alice graduated from Harvard, etc etc.  That's all great and extremely useful.  But when we extend Alice's record to say "Alice can do this" via VCs that *is* creating an Access Control List.  ACLs are about authority by identity and ocaps are about authority by possession.  These are not equivalent, and this misunderstanding even has a name as the "equivalence myth" as discussed in [Capability Myths Demolished](http://srl.cs.jhu.edu/pubs/SRL2003-02.pdf).  For more on why ACLs fall apart, the paper [ACLs Don't](http://waterken.sourceforge.net/aclsdont/current.pdf) goes into detail.

These are not functionally the same.  Verifiable Credentials can talk about all sorts of things, and does not specify an invocation algorithm, and its termsOfUse "attenuation" *can be* purely advisory.  ocap-ld as one particular implementation of ocaps can only talk about *one thing*, which is the granting of authority to some cryptographic material (note I say cryptographic material, not entity, since this is merely the mechanism by which we activate the "invokeable possession" of the capability), and it does have a very specific invocation algorithm that either succeeds or fails.  We can't use ocap-ld to talk about Alice's eyes and it's actually critical that we can't because we aren't really talking about Alice, we're instead putting some capability into her possession.  This difference is important. 

On the difference between ACLs and ocaps, which I fully acknowledge is difficult to perceive (and ocap literature is a heck of a slog to go through), I think [the explaination of capabilities in "E in a Walnut"](http://www.skyhunter.com/marcs/ewalnut.html#SEC42) explains it well (ignoring whether or not it's a good idea to have guns in your house at all especially with children around):

> **Principle of Least Authority (POLA)**
>
> When you buy a gallon of milk at the local 7-11, do you hand the cashier your wallet and say, "take what you want and give me the rest back?" Of course not. In handing the cashier exact change rather than your wallet, you are using the Principle of Least Authority, or POLA: you are giving the cashier no more authority than he needs. [...]
>
> **Children with your ID badge**
>
> Suppose all security in the physical world were based on ID badges and ID readers. At your home you might put an ID reader on your door, another on your CD cabinet, and another on your gun vault. Suppose further you had to depend on 4-year-old children to fetch your CDs for you when you were at the office. How would you do it? You would hand your ID badge to the child, and the child could then go through the front door and get into the CD cabinet. Of course, the child with your ID badge could also go into the gun vault. Most of the children would most of the time go to the CD cabinet, but once in a while one would pick up a gun, with lamentable results.
>
> **Keys**
>
> In the real physical world, if you had to depend on children to fetch CDs, you would not use an ID badge. Instead you would use keys. You would give the child a key to the front door, and a key to the CD cabinet. You would not give the child a key to the gun vault.
>
> [...]
>
> The "children" are the applications we run. In blissful unawareness, we give our ID badges to the programs automatically when we start them. The CD cabinet is the data a particular application should work on. The gun vault is the sensitive data to which that particular application should absolutely not have access. The children that always run to get a gun are computer viruses like the Love Bug.
>
> In computerese, ID badge readers are called "access control lists". Keys are called "capabilities". The basic idea of capability security is to bring the revolutionary concept of an ordinary door key to computing.

Reply:
Here is another, less gun-involving metaphor, which may or may not have been based on a real scenario (I was told it at Rebooting Web of Trust):

Alan's garage was under construction, so he needed to move his car someplace.  He asked Mark if he could store it in his garage, and Mark unhesitatingly said sure.  Mark handed his garage key to his son, and Mark's son handed it to Alan, and Alan stored his car in Mark's garage.  Later, Alan wanted to store a file on Mark's computer, and asked Mark to give him his password so he could upload the file.  Mark immediately objected uncomfortably to the idea.  Why?

- As an exercise for the reader, what does Mark's reaction in both cases tell us about ACLs versus ocaps?
- As a bonus exercise, let's say we have the same scenario but instead let's say that Alan wants to store a file on Mark's employer's file storage service, of which Mark has access (but is forbidden to allow non-employees to upload to his account).  In this scenario Mark does give Alan his corporate password, and Alan successfully logs in and uploads a file (and, we hope, does nothing else naughty).  What does this tell us about forbidding delegation being merely advisory?

Reply:
@cwebber I think you are conflating two different things: concepts and formats. There are two different access control concepts, namely: capabilities and access control lists, but they are both still credentials/permissions to do something. There are two different linked data formats currently being proposed: one in the VC spec and one in the OCAP spec. Both formats can be used to store both access control concepts. This is because of the open world linked data assumption, since both formats are built with object properties that are infinitely extensible. I am arguing that only one format is needed, since it can be used to store both concepts. When we remove the subject id from a VC, we have a capability. When we add a subject id to an OCAP we have an ACL. The former has already been done in the VC spec (bearer credentials) and the latter can easily be done by anyone extending the OCAP structure. We have already had lots of discussions about extensibility in the VC WG, and it has often been said that you cannot stop anyone extending the VC structure in any way they wish. So presumably this also applies to the OCAP structure as well. I dont see anything in your current spec that forbids this.

P.S. giving someone your username and password is not delegation, it is masquerade, which is entirely different to delegation.

Reply:
Responded to your P.S. [here](https://github.com/w3c/vc-data-model/pull/169/files#r186262245) so we can keep this thread about the ocap-ld vs VC stuff.

Now!  Actually you and I are maybe in more agreement than may be obvious:

> When we remove the subject id from a VC, we have a capability. When we add a subject id to an OCAP we have an ACL. The former has already been done in the VC spec (bearer credentials)

Okay, when it comes to bearer credentials I agree-ish!  Without the subject, this does resemble other bearer ocaps.  Let's leave aside whether or not it *should* be a bearer ocap (some discussion about the implications of delegation there, and maybe some other aspects about whether it's a good fit in other ways) but the thing that makes this not dangerous for authority is that we've removed discussion about identity altogether from this credential.  There's no risk AFAICT of it being an ACL.

Here's the thing though!  ocap-ld supports non-bearer capabilities!  You have the full power of a bearer capability if you see it, and ocap-ld is built for things like DIDs and blockchains where the capabilities can be observed without really being "possessed" by other participants.  So we do add information about what entity has been granted possession of an ocap, but it's critical that this is only for use of the invocation algorithm (or possibly, after the fact, to observe if abuse has occurred and consider revoking).

> and the latter can easily be done by anyone extending the OCAP structure.

You just described the risk perfectly!  In the case of the non-bearer ocaps (let's say "certificate style ocaps") if we accidentally start interpreting that same field by which we talk about what material has been granted possession powers and we mix it up with a vocabulary where we "talk about" various entities and extend their record, we accidentally adopt an ACL.  Because ocaps should not talk about identity!  ld-ocap has a very specific algorithm that passes or fails and we don't care about "reasoning" about what kind of entity a person is.  It's very important to keep these separate, and that's why we need a separate vocabulary, because the algorithms that process these datastructures are different.

Does that make sense?

Reply:
I am trying to determine where the border between VCs and OCAPs lies. Here is a taxonomy of 6 different types of credential. Which should be encoded as VCs and which as OCAPS, and why? But first some definitions that I am using:

Credential: the token possessed by someone or something that gives them a certain privilege or permission
Permission = the ability to perform an action on an object e.g. drive a car, read a file, delegate a credential
Privilege = the indirect ability to perform an action on an object, by virtue of some other attribute e.g. a role, a position in society, an identity

Based on the above definitions, we have the following 6 types of credential (at least):
i) a bearer permission. This credential states the action and the object e.g. can drive car X. Anyone or anything that possesses this credential can perform the stated action on the stated object.
ii) an identity permission. This credential states (the identity of) the subject, the action and the object e.g. David can drive car X. Anyone who can prove they have the stated identity can perform the action on the object.
iii) a keyed permission. This credential states the cryptographic key material, the action and the object. e.g. Key ID Y can drive car X. Anyone or anything that has the key material Y can perform the action on the object.
iv) a bearer privilege. This credential states the attribute that the holder of the credential possesses. e.g. the bearer is the President of the US. The holder will inherit all the permissions of the President of the US.
v) an identity privilege. This credential states (the identity of) the subject and the attribute that the subject possesses e.g. David is the President of the US. Anyone who has this credential and can prove they are David will inherit the permissions of the presidency.
vi) a keyed privilege. This credential states the cryptographic key material, and the attribute that the holder of the credential possesses e.g. Key ID Y is President of the US.

Reply:
@David-Chadwick,

I'm sure others will have something to say here, but here's my initial attempt at breaking things down in a simple way...

An issuer gives you a VC. A verifier gives you an OCAP.

> Permission = the ability to perform an action on an object e.g. drive a car, read a file, delegate a credential

This is an OCAP.

> Privilege = the indirect ability to perform an action on an object, by virtue of some other attribute e.g. a role, a position in society, an identity

This is a VC with the caveat that the issuer doesn't get to decide what the privileges are -- they simply to make claims about the characteristics/attributes that an identity possesses. It is up to a verifier of the VC to make a decision as to what permissions (capabilities) to give out on that basis. From there, the capabilities can be used without needing the identity information any longer. An OCAP-verifier may require authentication in order to accept an OCAP, but it does not require identity. OCAPs may also be delegated without the use of identity -- again needing only authentication.

OCAPs are for doing things and VCs are for talking about things.

Reply:
this one still isnt resolved
ii) an identity permission. This credential states (the identity of) the subject, the action and the object e.g. David can drive car X. Anyone who can prove they have the stated identity can perform the action on the object.
@dlongley says permissions are ocap, but an OCAP does not require identity. 
The above does require identity, so what is it?